GameOfLife.java

```diff
@@ -84,31 +109,68 @@ private static int[][] evolve(int[][] board) {
 	// Assumes that i is at least 1 and at most the number of rows in the board - 1. 
 	// Assumes that j is at least 1 and at most the number of columns in the board - 1. 
 	// Uses the count(board,i,j) function to count the number of alive neighbors.
-	private static int cellValue(int[][] board, int i, int j) {
+	public static int cellValue(int[][] board, int i, int j) {
 		//// Replace the following statement with your code.
-		return 0;
+		int alive = count(board, i, j);
+		int value = board[i][j]; 
+		if (board[i][j] == 1){
+			if ((alive < 2) || (alive > 3)){
+				value = 0;
+			}
+		}
+		else if (board[i][j] == 0) {
+			if (alive == 3){
+				value = 1;
+			}
+		}
+		return value;
 	}
 	
 	// Counts and returns the number of living neighbors of the given cell
 	// (The cell itself is not counted).
 	// Assumes that i is at least 1 and at most the number of rows in the board - 1. 
 	// Assumes that j is at least 1 and at most the number of columns in the board - 1. 
-	private static int count(int[][] board, int i, int j) {
+	public static int count(int[][] board, int i, int j) {
 		//// Replace the following statement with your code.
-		return 0;
+		int count = 0; 
+		for (int k=j-1; k<=j+1; k++){
+			if (board[i-1][k]==1){
+				count ++;
+			}
+		}
+		if (board[i][j-1]==1){
+			count ++;
+		}
+		if (board[i][j+1]==1){
+			count ++;
+		}
+		for (int k=j-1; k<=j+1; k++){
+			if (board[i+1][k]==1){
+				count ++;
+			}
+		}
+		return count;
 	}
 	
 	// Prints the board. Alive and dead cells are printed as 1 and 0, respectively.
-    private static void print(int[][] arr) {
+    public static void print(int[][] arr) {
 		//// Write your code here.
+		for (int i=1; i<arr.length-1; i++){
+			for (int j=1; j<arr[i].length-1; j++){
+				System.out.printf("%3s", arr[i][j]);
+				if (j == arr[i].length-2){
+					System.out.println(); 
+				}
+			}
```

Comment:
Instead of checking each time either it the last character in line, it's better to just create a new line when you know you done for sure.
```suggestion
			}
			System.out.println(); 
```